src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -1236,4 +1238,101 @@ public ParticipationStatus getParticipationStatus( UUID studyId, String particip
 
         return ParticipationStatus.UNKNOWN;
     }
+
+    @Override
+    public ChronicleQuestionnaire getQuestionnaire( UUID studyId, UUID questionnaireEKID ) {
+        SearchApi searchApi;
+        EntitySetsApi entitySetsApi;
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            searchApi = apiClient.getSearchApi();
+            entitySetsApi = apiClient.getEntitySetsApi();
+        } catch ( ExecutionException e ) {
+            throw new IllegalStateException( "unable to get apis", e );
+        }
+
+        logger.info( "Retrieving questionnaire: studyId = {}, questionnaire EKID = {}", studyId, questionnaireEKID );
+
+        try {
+            UUID studyEKID = getStudyEntityKeyId( studyId );
+            if ( studyEKID == null ) {
+                throw new IllegalArgumentException( "invalid studyId: " + studyId );
+            }
+
+            String questionnaireESName = getQuestionnaireEntitySetName( studyId );
+            String partOfESName = getPartOfAssociationEntitySetName( studyId );
+            String questionsESNAME = getQuestionnaireQuestionsESName( studyId );
+
+            Map<String, UUID> entitySets = entitySetsApi
+                    .getEntitySetIds( Set.of( questionnaireESName, questionsESNAME, partOfESName ) );
+
+            // Get study neighbors in questionnaire entity set filtered by study EKID
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi.executeFilteredEntityNeighborSearch(
+                    studyESID,
+                    new EntityNeighborsFilter(
+                            Set.of( studyEKID ),
+                            java.util.Optional.of( Set.of( entitySets.get( questionnaireESName ) ) ),
+                            java.util.Optional.of( Set.of( studyESID ) ),
+                            java.util.Optional.of( Set.of( entitySets.get( partOfESName ) ) )
+
+                    )
+            );
+            ChronicleQuestionnaire questionnaire = new ChronicleQuestionnaire();
+            if ( neighbors.containsKey( studyEKID ) ) {
+
+                // find questionnaire entity matching given entity key id
+                neighbors.get( studyEKID )
+                        .stream()
+                        .filter( neighbor -> neighbor.getNeighborDetails().isPresent() && neighbor.getNeighborId()
```

Comment:
Isn't this guaranteed to be present?

Reply:
I would expect it's always present :)  However, it's typed as Optional and I think it's better to assume that there is a chance it's not present.